<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>799</width>
    <height>529</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Automatic ROM-Core testing tool</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QHBoxLayout" name="horizontalLayout">
    <item>
     <widget class="QGroupBox" name="groupRoms">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
        <horstretch>3</horstretch>
        <verstretch>1</verstretch>
       </sizepolicy>
      </property>
      <property name="title">
       <string>ROM queue</string>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout_3">
       <item>
        <widget class="QListWidget" name="listRoms">
         <property name="frameShadow">
          <enum>QFrame::Sunken</enum>
         </property>
         <property name="horizontalScrollBarPolicy">
          <enum>Qt::ScrollBarAlwaysOff</enum>
         </property>
         <property name="editTriggers">
          <set>QAbstractItemView::NoEditTriggers</set>
         </property>
         <property name="dragEnabled">
          <bool>true</bool>
         </property>
         <property name="dragDropMode">
          <enum>QAbstractItemView::DropOnly</enum>
         </property>
         <property name="defaultDropAction">
          <enum>Qt::MoveAction</enum>
         </property>
         <property name="alternatingRowColors">
          <bool>true</bool>
         </property>
         <property name="selectionMode">
          <enum>QAbstractItemView::MultiSelection</enum>
         </property>
         <property name="selectionBehavior">
          <enum>QAbstractItemView::SelectRows</enum>
         </property>
         <property name="movement">
          <enum>QListView::Static</enum>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QFrame" name="romButtonContainer">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
           <horstretch>1</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="minimumSize">
          <size>
           <width>0</width>
           <height>50</height>
          </size>
         </property>
         <property name="maximumSize">
          <size>
           <width>16777215</width>
           <height>50</height>
          </size>
         </property>
         <layout class="QHBoxLayout" name="romButtonsLayout" stretch="1,0,0">
          <property name="leftMargin">
           <number>1</number>
          </property>
          <item>
           <widget class="QPushButton" name="buttonAddRom">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
              <horstretch>1</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="text">
             <string>Add</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="buttonRemoveRom">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
              <horstretch>1</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="text">
             <string>Remove</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="buttonClearRom">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
              <horstretch>1</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="text">
             <string>Clear</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
    <item>
     <widget class="QGroupBox" name="groupCores">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
        <horstretch>3</horstretch>
        <verstretch>1</verstretch>
       </sizepolicy>
      </property>
      <property name="title">
       <string>Core queue</string>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout_4">
       <item>
        <widget class="QListWidget" name="listCores">
         <property name="frameShadow">
          <enum>QFrame::Sunken</enum>
         </property>
         <property name="horizontalScrollBarPolicy">
          <enum>Qt::ScrollBarAlwaysOff</enum>
         </property>
         <property name="editTriggers">
          <set>QAbstractItemView::NoEditTriggers</set>
         </property>
         <property name="dragEnabled">
          <bool>true</bool>
         </property>
         <property name="dragDropMode">
          <enum>QAbstractItemView::DropOnly</enum>
         </property>
         <property name="defaultDropAction">
          <enum>Qt::MoveAction</enum>
         </property>
         <property name="alternatingRowColors">
          <bool>true</bool>
         </property>
         <property name="selectionMode">
          <enum>QAbstractItemView::MultiSelection</enum>
         </property>
         <property name="selectionBehavior">
          <enum>QAbstractItemView::SelectRows</enum>
         </property>
         <property name="movement">
          <enum>QListView::Static</enum>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QFrame" name="coreButtonContainer">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
           <horstretch>1</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="minimumSize">
          <size>
           <width>0</width>
           <height>50</height>
          </size>
         </property>
         <property name="maximumSize">
          <size>
           <width>16777215</width>
           <height>50</height>
          </size>
         </property>
         <layout class="QHBoxLayout" name="romButtonsLayout_2" stretch="1,0,0">
          <property name="leftMargin">
           <number>1</number>
          </property>
          <item>
           <widget class="QPushButton" name="buttonAddCore">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
              <horstretch>1</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="text">
             <string>Add</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="buttonRemoveCore">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
              <horstretch>1</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="text">
             <string>Remove</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="buttonClearCore">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
              <horstretch>1</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="text">
             <string>Clear</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
    <item>
     <widget class="QGroupBox" name="groupBox">
      <property name="enabled">
       <bool>true</bool>
      </property>
      <property name="sizePolicy">
       <sizepolicy hsizetype="Maximum" vsizetype="Preferred">
        <horstretch>2</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="minimumSize">
       <size>
        <width>50</width>
        <height>0</height>
       </size>
      </property>
      <property name="maximumSize">
       <size>
        <width>200</width>
        <height>16777215</height>
       </size>
      </property>
      <property name="title">
       <string>Test controls</string>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout_2">
       <item>
        <widget class="QPushButton" name="buttonStartTests">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Minimum" vsizetype="Expanding">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="text">
          <string>Start tests</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="buttonPauseResume">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Minimum" vsizetype="Expanding">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="text">
          <string>Pause/resume</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="buttonAbortTests">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Minimum" vsizetype="Expanding">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="text">
          <string>Abort tests</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="labelStatusRom">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="minimumSize">
          <size>
           <width>0</width>
           <height>25</height>
          </size>
         </property>
         <property name="maximumSize">
          <size>
           <width>16777215</width>
           <height>25</height>
          </size>
         </property>
         <property name="text">
          <string>Current ROM: 1942.zip</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="labelStatusCore">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="minimumSize">
          <size>
           <width>0</width>
           <height>25</height>
          </size>
         </property>
         <property name="maximumSize">
          <size>
           <width>16777215</width>
           <height>25</height>
          </size>
         </property>
         <property name="text">
          <string>Current core: MAME Current</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QGroupBox" name="boxResults">
         <property name="title">
          <string>Results</string>
         </property>
         <layout class="QVBoxLayout" name="verticalLayout">
          <item>
           <widget class="QLabel" name="labelTestLoad">
            <property name="text">
             <string>Load... OK</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="labelTestVideo">
            <property name="text">
             <string>Video... OK</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="labelTestControls">
            <property name="text">
             <string>Controls... OK</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="labelTestAudio">
            <property name="text">
             <string>Audio... OK</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="labelStatusOverview">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="minimumSize">
          <size>
           <width>0</width>
           <height>25</height>
          </size>
         </property>
         <property name="maximumSize">
          <size>
           <width>16777215</width>
           <height>25</height>
          </size>
         </property>
         <property name="text">
          <string>Running... 12/8192</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QProgressBar" name="progressBar">
         <property name="value">
          <number>24</number>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>799</width>
     <height>21</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <widget class="QMenu" name="menuExport">
     <property name="title">
      <string>Export...</string>
     </property>
     <addaction name="actionExportTextile"/>
     <addaction name="actionExportPineapple"/>
    </widget>
    <addaction name="actionNew"/>
    <addaction name="actionLoad"/>
    <addaction name="actionSave"/>
    <addaction name="menuExport"/>
    <addaction name="separator"/>
    <addaction name="actionExit"/>
   </widget>
   <widget class="QMenu" name="menuTests">
    <property name="title">
     <string>Tests</string>
    </property>
    <addaction name="actionAddCore"/>
    <addaction name="actionAddRom"/>
    <addaction name="separator"/>
    <addaction name="actionStartTests"/>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menuTests"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <action name="actionExportTextile">
   <property name="text">
    <string>RedmineTextile</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+E</string>
   </property>
  </action>
  <action name="actionExportPineapple">
   <property name="text">
    <string>Pineapple</string>
   </property>
  </action>
  <action name="actionNew">
   <property name="text">
    <string>New set</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+N</string>
   </property>
  </action>
  <action name="actionSave">
   <property name="text">
    <string>Save set...</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+S</string>
   </property>
  </action>
  <action name="actionLoad">
   <property name="text">
    <string>Load set...</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+O</string>
   </property>
  </action>
  <action name="actionAddCore">
   <property name="text">
    <string>Add Core(s)</string>
   </property>
  </action>
  <action name="actionAddRom">
   <property name="text">
    <string>Add ROM(s)</string>
   </property>
  </action>
  <action name="actionExit">
   <property name="text">
    <string>Exit</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Q</string>
   </property>
  </action>
  <action name="actionStartTests">
   <property name="text">
    <string>Begin tests</string>
   </property>
  </action>
 </widget>
 <resources/>
 <connections>
  <connection>
   <sender>actionExit</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>close()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>264</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>buttonAddCore</sender>
   <signal>clicked()</signal>
   <receiver>actionAddCore</receiver>
   <slot>trigger()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>354</x>
     <y>464</y>
    </hint>
    <hint type="destinationlabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>buttonAddRom</sender>
   <signal>clicked()</signal>
   <receiver>actionAddRom</receiver>
   <slot>trigger()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>60</x>
     <y>464</y>
    </hint>
    <hint type="destinationlabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>buttonStartTests</sender>
   <signal>clicked()</signal>
   <receiver>actionStartTests</receiver>
   <slot>trigger()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>693</x>
     <y>85</y>
    </hint>
    <hint type="destinationlabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>buttonClearCore</sender>
   <signal>clicked()</signal>
   <receiver>listCores</receiver>
   <slot>clear()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>531</x>
     <y>464</y>
    </hint>
    <hint type="destinationlabel">
     <x>447</x>
     <y>243</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>buttonClearRom</sender>
   <signal>clicked()</signal>
   <receiver>listRoms</receiver>
   <slot>clear()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>236</x>
     <y>464</y>
    </hint>
    <hint type="destinationlabel">
     <x>152</x>
     <y>243</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
